expression/builtin_cast.go

```diff
@@ -726,8 +726,7 @@ func (b *builtinCastRealAsDurationSig) evalDuration(row []types.Datum) (res type
 	if isNull || err != nil {
 		return res, isNull, errors.Trace(err)
 	}
-	res, err = types.ParseDuration(strconv.FormatFloat(val, 'f', -1, 64), b.tp.Decimal)
-	return res, false, errors.Trace(err)
+	return types.ParseDuration(strconv.FormatFloat(val, 'f', -1, 64), b.tp.Decimal)
```

Comment:
```go
res, err = types.ParseDuration(strconv.FormatFloat(val, 'f', -1, 64), b.tp.Decimal)
return res, err != nil, errors.Trace(err)
```

Reply:
I modified func ParseDuration to return isNull param.

Reply:
But we still should need to `Trace` the error when it is not `nil`.